configure

```diff
@@ -2,9 +2,47 @@
 
 . scripts/configure.sh || exit 1
 
+c11_code="
+#include <stdatomic.h>
+
+int main(void)
+{
+#ifdef __STDC_NO_ATOMICS__
+#error No C11 atomics
+#endif
+	_Atomic int res = ATOMIC_VAR_INIT(0);
+
+	return res;
+}
+"
+
+check_c11()
+{
+	msg_checking "for C11 (with atomics support)"
+
+	for flag in -std=gnu11 -std=gnu1x -std=c11 ""
```

Comment:
I was thinking of potential compilers where gnu11 isn't available, but other C11 options (or none at all) would be enough to build cmus.

I don't think these fallbacks hurt, unless we really only want gnu11 compatibility (I could be fine with that).